managed/devops/opscli/ybops/cloud/aws/cloud.py

```diff
@@ -381,11 +382,24 @@ def stop_instance(self, args):
         except ClientError as e:
             logging.error(e)
 
-    def start_instance(self, args):
+    def start_instance(self, args, ssh_port):
         ec2 = boto3.resource('ec2',  args["region"])
         try:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
             instance = ec2.Instance(id=args["id"])
             instance.start()
             instance.wait_until_running()
+            # The OS boot up may take some time,
+            # so retry until the instance allows SSH connection.
+            retry_count = 0
+            while retry_count < 30:
```

Comment:
What happens if even after 30 tries the instance isn't up? We should throw an error here saying instance not reachable or similar.

Reply:
Raised exception also added log. `Logging.error `is fine here? or should we add a log in info?